haproxy_discovery.sh

```diff
@@ -11,7 +11,7 @@
 #  global
 #  stats socket /run/haproxy/info.sock  mode 666 level user
 
-HAPROXY_SOCK=$(echo $1 | grep -Po "^/[^\s]+" || echo "/var/run/haproxy/info.sock")
+HAPROXY_SOCK=$(echo $1 | tr -d '\040\011\012\015' || echo "/var/run/haproxy/info.sock")
```

Comment:
unfortunately, `tr ` doesn't return non-zero error code when pattern doesn't match :( so it breaks the functionality when no explicit HAPROXY_SOCK is passed as first argument.. 
we'll have to do it a bit differently here if we want to use `tr`
how about this? does this work for you?
```
HAPROXY_SOCK="/var/run/haproxy/info.sock"
[ -n "$1" ] && echo $1 | grep -q ^/ && HAPROXY_SOCK="$(echo $1 | tr -d '\040\011\012\015')"
```